staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -216,7 +215,9 @@ func (u *udsGRPCConnector) connect(_ context.Context) (proxier, error) {
 	// we cannot use ctx just for dialing and control the connection lifetime separately.
 	// See https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/357.
 	tunnelCtx := context.TODO()
-	tunnel, err := client.CreateSingleUseGrpcTunnel(tunnelCtx, udsName, dialOption,
+	tunnel, err := client.CreateSingleUseGrpcTunnelWithContext(connectCtx, tunnelCtx, udsName, dialOption,
+		grpc.WithBlock(),
+		grpc.FailOnNonTempDialError(true),
```

Comment:
Ah, I think we might want `grpc.WithReturnConnectionError()` instead

https://github.com/kubernetes/kubernetes/blob/master/vendor/google.golang.org/grpc/dialoptions.go#L273-L287

Reply:
Which is unfortunately also experimental

Reply:
Agreed looks positive, included.